arangod/Cluster/ClusterInfo.cpp

```diff
@@ -6215,19 +6215,61 @@ ClusterInfo::getResponsibleServerReplication2(std::string_view shardID) {
     }
   }
 
-  auto logId = LogicalCollection::shardIdToStateId(shardID);
   auto result = std::shared_ptr<std::vector<ServerID>>{nullptr};
 
   while (true) {
     {
+      if (++tries >= 100 || _server.isStopping()) {
+        break;
+      }
+
       READ_LOCKER(readLocker, _planProt.lock);
-      // if we find a replicated log for this shard, then this is a
-      // replication 2.0 db, in which case we want to use the participant
-      // information from the log instead
+
+      // TODO we have to simplify this by using some adequate data structures
+      // lookup the collection name
+      auto colName = _shardToName.find(shardID);
+      if (colName == _shardToName.end()) {
+        continue;
+      }
+
+      // fetch the logical collection containing the shard
+      std::shared_ptr<LogicalCollection> logical{nullptr};
+      for (auto const& [dbID, col] : _plannedCollections) {
```

Comment:
Wait a second, this sounds too complicated...

Can we talk about this in person?
We should be able to identify the group & shardSheaf index for a a shard much easier.

Reply:
I did not find an easier way to do that and created a ticket for simplifying this in the future: [CINFRA-700](https://arangodb.atlassian.net/browse/CINFRA-700). I think we're missing some of the necessary structures in `ClusterInfo`, but I'll gladly change it if there's an easier workaround.